internal/flowrate/xrate/monitor.go

```diff
@@ -0,0 +1,98 @@
+package xrate
+
+import (
+	"time"
+
+	"golang.org/x/time/rate"
+
+	legacyimpl "github.com/cometbft/cometbft/internal/flowrate/legacy"
+	cmtsync "github.com/cometbft/cometbft/libs/sync"
+)
+
+type Monitor struct {
+	limiter *rate.Limiter
+
+	// rate.Limiter configuration
+	burst int
+	rate  int64
+
+	// Tokens reserved but not yet used
+	reserved int
+
+	statusMtx cmtsync.Mutex
+	status    legacyimpl.Status
+}
+
+func New(_, _ time.Duration) *Monitor {
+	return &Monitor{
+		limiter: rate.NewLimiter(rate.Inf, 0),
+	}
+}
+
+func (m *Monitor) Limit(want int, maxRate int64, block bool) (n int) {
```

Comment:
I struggle to see what this method would be useful for if `block` is set to false...

Reply:
We never used it with `false` actually :)